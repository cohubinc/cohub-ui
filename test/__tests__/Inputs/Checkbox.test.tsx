import React from "react";
import { mount } from "enzyme";

import { Inputs, StoryHelpers } from "dist";
import { fireEvent } from "@testing-library/dom";
import { render } from "@testing-library/react";

describe("CheckBox Input", () => {
  it("renders checked without crashing", async () => {
    const wrapper = mount(
      <Inputs.Checkbox
        label="checkBox"
        input={{ value: true } as any}
        meta={{}}
        id="checkbox_input"
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("renders un-checked without crashing", async () => {
    const wrapper = mount(
      <Inputs.Checkbox
        label="checkBox"
        input={{ value: false } as any}
        meta={{}}
        id="checkbox_input"
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("toggles on click", () => {
    const { getByLabelText } = render(
      <StoryHelpers.StateContainer defaultState="">
        {({ state, setState }) => (
          <Inputs.Checkbox
            label="Checkbox Input"
            key="selected"
            id="checkbox_input"
            input={{
              name: "selected",
              value: state,
              onChange: () => setState(!state)
            }}
          />
        )}
      </StoryHelpers.StateContainer>
    );

    expect(getByLabelText("Checkbox Input")).toHaveAttribute(
      "data-checked",
      "false"
    );

    fireEvent.click(getByLabelText("Checkbox Input"));

    expect(getByLabelText("Checkbox Input")).toHaveAttribute(
      "data-checked",
      "true"
    );
  });
});
