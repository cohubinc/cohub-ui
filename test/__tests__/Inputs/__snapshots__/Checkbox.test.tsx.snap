// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox Input renders checked without crashing 1`] = `
<Checkbox
  id="checkbox_input"
  input={
    Object {
      "value": true,
    }
  }
  label="checkBox"
>
  <div
    className="CheckboxField-Container flex justify-start items-center"
  >
    <div
      className="CheckboxField mr-05"
      data-checked={true}
      id="checkbox_input"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="checkbox"
      tabIndex={0}
    >
      <Component
        color="hsl(114, 35%, 52%)"
        size={16}
      >
        <DefaultIcon
          color="hsl(114, 35%, 52%)"
          name="checkmark"
          size={16}
        >
          <Checkmark
            color="hsl(114, 35%, 52%)"
            name="checkmark"
            size={16}
          >
            <IconWrapper
              color="hsl(114, 35%, 52%)"
              name="checkmark"
              size={16}
            >
              <div
                className="CohubIcon checkmarkIcon  "
                style={
                  Object {
                    "height": 16,
                    "opacity": undefined,
                    "transition": "opacity 300ms ease-in 200ms",
                    "width": 16,
                  }
                }
              >
                <div
                  className="flex justify-center items-center"
                >
                  <svg
                    fill="none"
                    height={16}
                    viewBox="0 0 24 24"
                    width={16}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M9 16.17L4.83 12L3.41 13.41L9 19L21 7L19.59 5.59L9 16.17Z"
                      fill="hsl(114, 35%, 52%)"
                    />
                  </svg>
                </div>
              </div>
            </IconWrapper>
          </Checkmark>
        </DefaultIcon>
      </Component>
    </div>
    <label
      htmlFor="checkbox_input"
    >
      checkBox
    </label>
  </div>
</Checkbox>
`;

exports[`CheckBox Input renders un-checked without crashing 1`] = `
<Checkbox
  id="checkbox_input"
  input={
    Object {
      "value": false,
    }
  }
  label="checkBox"
>
  <div
    className="CheckboxField-Container flex justify-start items-center"
  >
    <div
      className="CheckboxField mr-05"
      data-checked={false}
      id="checkbox_input"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="checkbox"
      tabIndex={0}
    />
    <label
      htmlFor="checkbox_input"
    >
      checkBox
    </label>
  </div>
</Checkbox>
`;
